### PR TITLE
fix(typing) Add unreachable flag to mypy

### PR DESCRIPTION
**Pull request by evanh**
Add the unreachable flag to mypy so that we make sure there aren't code sections
that mypy is skipping.

There are a few errors fixed in https://github.com/getsentry/snuba/pull/1713 that need to be merged first for the CI to pass here.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -25,6 +25,6 @@ repos:
     rev: 'v0.800'
     hooks:
     -   id: mypy
-        args: [--config-file, mypy.ini, --ignore-missing-imports, --strict]
+        args: [--config-file, mypy.ini, --ignore-missing-imports, --strict, --warn-unreachable]
```
Comment (lynnagara) [inline .pre-commit-config.yaml:28]: Not related to your change, but do we actually need `--ignore-missing-imports`? AFAIK the information about which specific imports to ignore is specified in mypy.ini.

**Review by lynnagara — APPROVED** [review]

**Comment by lynnagara** [comment]
ok I'm confused.. how is it that ci is passing already?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (evanh) [reply .pre-commit-config.yaml:28]: I can't remember if I put that there for a reason or I just copied it from somewhere else. I'd have to test it.

**Comment by evanh** [comment]
I don't think the CI is running the new actions yet. Which worries me because I don't know how to test them. :(

**Comment by evanh** [comment]
> ok I'm confused.. how is it that ci is passing already?

OK I have this working now. This should start passing once the transaction processor typing stuff is merged.

**Pull request merged**